src_c/rwobject.c

```diff
@@ -722,22 +737,43 @@ _rwops_from_pystr(PyObject *obj)
     if (obj != NULL) {
         SDL_RWops *rw = NULL;
         PyObject *oencoded;
+        char* encoded = NULL;
+        char* ext = NULL;
+        char* extension = NULL;
+
+        //longest extension is 4 chars, plus NULL at end
+        extension = malloc(sizeof(char)*5);
+        if (extension == NULL) {
+            return (SDL_RWops*)PyErr_NoMemory();
+        }
```

Comment:
hmmmm this part smells like a potential segfault magnet. By allocating only 5 bytes, we are not handling any wider extensions (and those do exist, just they are not common enough). Now, when a user tries to load such a file with a large extension, and exceeds 5 bytes, the `strcpy` will probably try to copy that into this small allocated memory space, and the rest of the data overflows and potentially segfaults

A cleaner solution here is to have just enough memory allocated to store the entire extension string (with some API functions of PyUnicode/PyString)

Reply:
Thanks for the feedback, I'll try and implement that.